custom-metrics-stackdriver-adapter/examples/prometheus-to-sd/README.md

```diff
@@ -0,0 +1,167 @@
+# Custom Metrics with Prometheus and Stackdriver
+
+This is a minimal example showing how to export custom metrics using 
+[prometheus text format](https://prometheus.io/docs/instrumenting/exposition_formats/) to Stackdriver
+in a k8s cluster running on GCE or GKE. 
+
+## Prometheus dummy exporter
+
+A simple prometheus-dummy-exporter component exposes a single prometheus metric 
+with a constant value. The metric name and value can be passed in via flags.
+There is also a flag to configure the port on which the metrics are served. 
+
+## Prometheus to Stackdriver sidecar container
+
+Prometheus Dummy exporter is deployed with a sidecar container running 
+[Prometheus to Stackdriver] container configured to scrape exposed metric from the specified port
+and write it to Stackdriver. [Custom metrics - Stackdriver adapter] then reads this metric from Stackdriver 
+and makes the metric available for scaling.
+
+### prometheus-to-sd flags
+Note important configuration details for prometheus-to-sd
+```yaml	
+command:
+  - /monitor
+  - --source=:http://localhost:8080  # Empty component name. Needed to avoid slashes in the custom metric name. Port number has to match the one configured for prometheus_dummy_exporter.
+  - --stackdriver-prefix=custom.googleapis.com   # Prefix needed for custom metrics
+  # pod-id and namespace-id are needed for the metrics exported to Stackdriver
+  # to be read correctly by custom-metrics-stackdriver-adapter.
+  - --pod-id=$(POD_ID)
+  - --namespace-id=$(POD_NAMESPACE)
+env:
+  - name: POD_ID
+    valueFrom:
+      fieldRef:
+        apiVersion: v1
+        fieldPath: metadata.uid
+  - name: POD_NAMESPACE
+    valueFrom:
+      fieldRef:
+        fieldPath: metadata.namespace
+```
+## Horizontal Pod Autoscaling object
+
+In the example, there is a [Horizontal Pod Autoscaler object] configured to scale the deployment on 
+the exported metric. It is configured with a target value lower than the defaul metric value. This will
+cause the deployment to scale up to 5 replicas over the course of ~15 minutes. 
+If you do not see a successful scale up event for a longer period of time, there is 
+likely an issue with your setup. See [Troubleshooting](#troubleshoooting) section for possible reasons.
+
+## Deployment
+
+The example can be used to test your custom metrics setup using Stackdriver. See the [custom metrics setup instructions] for more details.
```

Comment:
I feel this link should be more exposed and it should be clearly stated it's a prerequisite for this example.

Reply:
Very valid point. I mention this now in the Overview as a prerequisite.